website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,490 @@
+
+
+:::note
+
+The `DBT_CLOUD_GIT_BRANCH` variable is only available within the dbt Cloud IDE and not the Cloud CLI.
+
+:::
+
+
+We’ve also seen some organizations prefer to organize their dev databases by branch name. This requires implementing similar logic in `generate_database_name()` instead of the `generate_schema_name()` macro. By default, dbt will not automatically create the databases. You can refer to the [Tips and tricks](/guides/customize-schema-alias?step=4#tips-and-tricks) section on the next page to learn more.    
+
+
+```Jinja
+
+{% macro generate_schema_name(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if  env_var('DBT_ENV_TYPE','DEV') == 'DEV' -%}
+    
+        {#- we replace characters not allowed in the schema names by "_" -#}
+        {%- set re = modules.re -%}
+        {%- set cleaned_branch = re.sub("\W", "_", env_var('DBT_CLOUD_GIT_BRANCH')) -%}
+        
+        {%- if custom_schema_name is none -%}
+
+            {{ cleaned_branch }}
+
+        {%- else -%}
+
+             {{ cleaned_branch }}_{{ custom_schema_name | trim }}
+
+        {%- endif -%}
+        
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+```
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Branch      |Target database| Target schema | Resulting object                  |
+|-------------|:----------:|:-------------:|:-------------:|:---------------------------------:|
+| Developer 1 |`featureABC`|dev            | dbt_dev1      |dev.featureABC_marketing.my_model  |
+| Developer 2 |`featureABC`|dev            | dbt_dev2      |dev.featureABC_marketing.my_model  |
+| Developer 1 |`feature123`|dev            | dbt_dev1      |dev.feature123_marketing.my_model  |
+| CI PR 123   |            |ci             | dbt_pr_123    |ci.dbt_pr_123_marketing.my_model   |
+| CI PR 234   |            |ci             | dbt_pr_234    |ci.dbt_pr_234_marketing.my_model   |
+| Production  |            |prod           | analytics     |prod.marketing.my_model\           |
+
+
+When developer 1 and developer 2 are checked out on the same branch, they will generate the same object in the data warehouse, but this should not be a problem as being on the same branch means that the code of the model will be the same for the two developers.
+
+
+### 4. Use a static schema for CI 
+
+Some organizations prefer to write their CI jobs to a single schema with the PR identifier prefixed to the front of the table name. It's important to note that this will result in long table names. 
+
+
```

Comment:
i think this is missing instructions or something? we should tell the user to add these in their macros folder. Is it a separate file for each macro? i think so yea?

e.g.:
Create a new file called `generate_schema_name.sql` under your macros folder with the following code: